app/src/main/java/com/winlator/core/FileUtils.java

```diff
@@ -439,4 +467,131 @@ public static String readZipManifestNameFromAssets(Context context, String asset
         }
         return null;
     }
+
+    public static String readZipManifestNameFromInputStream(InputStream inStream) {
+        try (ZipInputStream zis = new ZipInputStream(inStream)) {
+            ZipEntry entry = zis.getNextEntry();
+            while (entry != null) {
+                if (!entry.isDirectory() && entry.getName().endsWith(".json")) {
+                    byte[] bytes = StreamUtils.copyToByteArray(zis);
+                    try {
+                        org.json.JSONObject json = new org.json.JSONObject(new String(bytes));
+                        String name = json.optString("name", json.optString("libraryName", "")).trim();
+                        if (!name.isEmpty()) {
+                            return name;
+                        }
+                    } catch (org.json.JSONException ignored) {
+                    }
+                }
+                zis.closeEntry();
+                entry = zis.getNextEntry();
+            }
+        } catch (IOException ignored) {
+        }
+        return null;
+    }
+
+    // --- High-level helpers for installed adrenotools drivers -------------------------------------------------
+    public static File getAdrenoInstalledRoot(Context context) {
+        return new File(context.getFilesDir(), "installed_components/adrenotools_driver");
+    }
+
+    public static List<String> listInstalledAdrenoDrivers(Context context) {
+        ArrayList<String> results = new ArrayList<>();
+        File root = getAdrenoInstalledRoot(context);
+        try {
+            if (root.isDirectory()) {
+                File[] dirs = root.listFiles((f) -> f.isDirectory());
+                if (dirs != null) {
+                    for (File d : dirs) results.add(d.getName());
+                }
+            }
+        } catch (Exception e) {
+            Log.e("FileUtils", "Failed to list Adreno drivers: " + e);
+        }
+        return results;
+    }
+
+    public static boolean deleteInstalledAdrenoDriver(Context context, String identifier) {
+        try {
+            File dir = new File(getAdrenoInstalledRoot(context), identifier);
+            if (dir.exists()) return delete(dir);
+            return false;
+        } catch (Exception e) {
+            return false;
+        }
+    }
```

Comment:
I'd slightly change this to early return pattern instead.

if (!dir.exists()) return false
....

return delete(dir);

Just keeps code consistency.